numpy/core/src/common/simd/vsx/arithmetic.h

```diff
@@ -118,6 +118,25 @@
 
 // Horizontal add: Calculates the sum of all vector elements.
 
+NPY_FINLINE npy_uint32 npyv_sum_u8(npyv_u8 a)
+{
+    const npyv_u32 zero4 = npyv_zero_u32();
+    npyv_u32 sum4 = vec_sum4s(a, zero4);
+    return (npy_uint32)vec_extract(vec_sums(sum4, zero4), 3);
+}
```

Comment:
```suggestion
NPY_FINLINE npy_uint16 npyv_sum_u8(npyv_u8 a)
{
    const npyv_u32 zero = npyv_zero_u32();
    npyv_u32 four = vec_sum4s(a, zero);
    npyv_u32 one  = vec_sums((npyv_s32)sum4, (npyv_s32)zero4);
    return (npy_uint16)vec_extract(one, 3);
}
```
cast to `npyv_s32` since `vec_sums` doesn't accept u32 vector.